spring-integration-stomp/src/main/java/org/springframework/integration/stomp/AbstractStompSessionManager.java

```diff
@@ -0,0 +1,231 @@
+import org.springframework.util.concurrent.ListenableFutureCallback;
+
+/**
+ * Base {@link StompSessionManager} implementation to manage the single {@link StompSession}
+ * over its {@link ListenableFuture} from the target implementation of this class.
+ * <p>
+ * The connect to {@link StompSession} is done during {@link #afterPropertiesSet()}.
+ * <p>
+ * The {@link #destroy()} lifecycle method manages {@link StompSession#disconnect()}.
+ * <p>
+ * The {@link #connect(StompSessionHandler)} and {@link #disconnect(StompSessionHandler)} method
+ * implementations populate/remove provided {@link StompSessionHandler} to/from an internal
+ * {@link AbstractStompSessionManager.CompositeStompSessionHandler}, which delegates all operations
+ * to the provided {@link StompSessionHandler}s.
+ * This {@link AbstractStompSessionManager.CompositeStompSessionHandler} is used for the
+ * {@link {@link StompSession} connection.
+ *
+ * @author Artem Bilan
+ * @since 4.2
+ */
+public abstract class AbstractStompSessionManager implements StompSessionManager, ApplicationEventPublisherAware,
+		InitializingBean, DisposableBean, BeanNameAware {
+
+	protected final Log logger = LogFactory.getLog(getClass());
+
+	private final CompositeStompSessionHandler compositeStompSessionHandler = new CompositeStompSessionHandler();
+
+	protected final StompClientSupport stompClient;
+
+	private ApplicationEventPublisher applicationEventPublisher;
+
+	private volatile StompHeaders connectHeaders;
+
+	private volatile ListenableFuture<StompSession> stompSessionListenableFuture;
+
+	private volatile boolean autoReceipt;
+
+	private volatile boolean connected;
+
+	private String name;
+
+	public AbstractStompSessionManager(StompClientSupport stompClient) {
+		Assert.notNull(stompClient, "'stompClient' is required.");
+		this.stompClient = stompClient;
+	}
+
+	public void setConnectHeaders(StompHeaders connectHeaders) {
+		this.connectHeaders = connectHeaders;
+	}
+
+	public void setAutoReceipt(boolean autoReceipt) {
+		this.autoReceipt = autoReceipt;
+	}
+
+	@Override
+	public boolean isAutoReceiptEnabled() {
+		return this.autoReceipt;
+	}
+
+	@Override
+	public boolean isConnected() {
+		return this.connected;
+	}
+
+	@Override
+	public void setApplicationEventPublisher(ApplicationEventPublisher applicationEventPublisher) {
+		this.applicationEventPublisher = applicationEventPublisher;
+	}
+
+	@Override
+	public void setBeanName(String name) {
+		this.name = name;
+	}
+
+	@Override
+	public void afterPropertiesSet() throws Exception {
+		this.stompSessionListenableFuture = doConnect(this.compositeStompSessionHandler);
+		this.stompSessionListenableFuture.addCallback(new ListenableFutureCallback<StompSession>() {
+
+			@Override
+			public void onFailure(Throwable e) {
+				logger.error("STOMP connect error.", e);
+				if (applicationEventPublisher != null) {
+					applicationEventPublisher.publishEvent(
+							new StompExceptionEvent(AbstractStompSessionManager.this, e));
+				}
```

Comment:
I think I agree here and this `StompSessionManager` really should take care about reconnect to the lost connection for the session. Like you do in the `MqttPahoMessageDrivenChannelAdapter`